samples/algorithms/BitFlip.qs

```diff
@@ -0,0 +1,139 @@
+/// # Sample
+/// Bit-Flip
+///
+/// # Description
+/// This sample demonstrates the three-qubit bit-flip code. This code is a
+/// simple quantum error correction strategy for protecting against a single
+/// bit-flip error by encoding a logical qubit into three physical qubits. A
+/// single bit-flip error is when one of the three physical qubits has its
+/// state changed erroneously in a way that is equivalent to applying the X
+/// gate to it.
+///
+/// The bit-flip correction code works by checking the parity of the physical
+/// qubits. By measuring only their parity, the quantum superposition of the
+/// qubits is preserved the qubits. Because all the physical qubits are
+/// supposed to have the same state, when the parity checks detect a
+/// difference in state, the erroneous qubit can be identified and corrected.
+///
+/// This Q# program prepares a logical qubit encoded as three physical qubits
+/// with one of the qubits being bit-flipped. It then identifies and corrects
+/// the flipped qubit.
+namespace Sample {
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Random;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Diagnostics;
+    open Microsoft.Quantum.Measurement;
+
+    @EntryPoint()
+    operation Main() : Result {
```

Comment:
I think having `physicalQubit`, `aux` and an alias for all of these qubits in an array (`logicalQubit`), makes the flow harder to follow.

I would suggest using a single 3-qubit register throughout main, and make `EncodeAsLogicalQubit` receive a qubit register rather than two separate arguments.

Reply:
I've gotten rid of the `physicalQubit` and `aux` aliases to just use `logicalQubit` throughout. It makes more sense to have the `EncodeAsLogicalQubit` take two arguments to distinguish the qubit that has the state to encode from the two qubits that will be used in the encoding. As they are functionally different, they should be separate arguments.